### PR TITLE
fix native assets outside of eth not working with preferred network

### PR DESCRIPTION
**Pull request by walmat**
## What changed (plus any additional context for devs)
Noticed that Apecoin wasn't being set as the inputAsset when the preferred network was set to apechain. This should fix it. Also renamed the function to be more specific, not just ETH but native token on that chain.

## Screen recordings / screenshots
https://github.com/user-attachments/assets/2024a70a-be61-4a51-9109-f28a6a14a9c3

## What to test
nothing really needs testing tbh

### CODE REVIEW

File: src/components/context-menu-buttons/ChainContextMenu.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const ChainContextMenu = ({
         actionTitle: chainsLabel[chainId],
         icon: {
           iconType: 'ASSET',
-          iconValue: `${chainsName[chainId]}Badge${chainId === ChainId.mainnet ? '' : 'NoShadow'}`,
+          iconValue: chainId === ChainId.mainnet ? 'ethereumBadge' : `${chainsName[chainId]}BadgeNoShadow`,
```
Comment (walmat) [inline src/components/context-menu-buttons/ChainContextMenu.tsx:64]: this was not showing mainnet badge

File: src/state/assets/userAssets.ts [hunk 2]
```diff
@@ -201,14 +201,14 @@ export const createUserAssetsStore = (address: Address | string) =>
...
-      getHighestValueEth: () => {
+      getHighestValueNativeAsset: () => {
         const preferredNetwork = swapsStore.getState().preferredNetwork;
         const assets = get().userAssets;
 
         let highestValueEth = null;
 
         for (const [, asset] of assets) {
-          if (asset.mainnetAddress !== ETH_ADDRESS) continue;
+          if (!asset.isNativeAsset) continue;
```
Comment (walmat) [inline src/state/assets/userAssets.ts:211]: here is the fix

**Review by greg-schrammel — APPROVED** [review]
nice

File: src/components/context-menu-buttons/ChainContextMenu.tsx (hunk 1 above)
Comment (greg-schrammel) [inline src/components/context-menu-buttons/ChainContextMenu.tsx:64]: do we have a `ethereumBadgeNoShadow` icon? if not we could change the icon name to keep a standard of being the chain name

**Review by jinchung — APPROVED** [review]
🌮

File: src/components/context-menu-buttons/ChainContextMenu.tsx (hunk 1 above)
Comment (walmat) [reply src/components/context-menu-buttons/ChainContextMenu.tsx:64]: we don't but I am planning on reworking these soon to be backend driven anyway so not going to bother on doing this now.

**Pull request merged**